ingest/clusters.py

```diff
@@ -73,7 +74,27 @@ def __init__(
     def validate(self):
         """ Runs all validation checks
         """
-        return all([self.is_valid_format(), self.validate_numeric_annots()])
+        return all(
+            [
+                self.is_valid_format(),
+                self.validate_numeric_annots(),
+                self.require_X_Y_not_nan(),
+            ]
+        )
+
+    def require_X_Y_not_nan(self):
+        """
+        X Y and Z are expected to be fully populated in cluster files
+        Fail coordinate data columns that have nan values
+        """
+        is_valid = True
+        for annot_name in self.headers:
+            if annot_name.lower() in ("x", "y", "z"):
+                if self.file[annot_name].isna().any().bool():
```

Comment:
Do we have to worry about an empty string here at all? It looks like `isna()`wouldn't count `''` as True here so just wanted to check (per these [docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.isna.html))

Reply:
Great question. We use Pandas to read in the data and any empty cells (a la  missing values in a spreadsheet) are converted to NaN. If the tsv file has literal quotes with nothing between, then ingest will hit a `could not convert string to float` ValueError. I'm not planning to catch that type of error gracefully because it seems hard to generate accidentally (having quotes in a column of numeric values) and also harder to overlook visually.

Thanks for the question as it prompted me to test a case I hadn't thought of!